6 weekly forecast component

phfernandesmc opened the pull request:
in this branch I created the context to share the city weather information for the whole app and I also created the weekly weather forecast component

phfernandesmc committed:
saving the progress for the weekly component

phfernandesmc committed:
context create

phfernandesmc committed:
weekly component done

phfernandesmc committed:
Delete src/components/test.js

Shingo023 commented on src/components/WeeklyComponent.tsx:
```diff
@@ -0,0 +1,70 @@
+  };
+
+  return (
+    <div className="WeeklyComponent">
```
Since we are going to use SCSS Modules for the project, the class name should be like `{styles.WeeklyComponent}`?
If so, you also need to create and import `module.scss` file to style your component.

Shingo023 commented on src/components/WeeklyComponent.tsx:
```diff
@@ -0,0 +1,70 @@
+export const WeeklyComponent = () => {
+  const { currentCity } = useCurrentCity();
+
+  const days: WeatherDay[] | undefined = currentCity?.days.slice(0, 7)
```
How about using 'weeklyWeather' in stead of 'days' as the variable name? I think that is more understandable.

Shingo023 commented on src/components/WeeklyComponent.tsx:
```diff
@@ -0,0 +1,70 @@
+      <h2>Weekly Forecast</h2>
+      <ul className="WeeklyComponentList">
+        {days ? (
+          days.map((day, index) => (
```
How about 'WeeklyWeather' instead of 'days' and 'dailyWeather' instead of 'day'?

Shingo023 commented on src/components/WeeklyComponent.tsx:
```diff
@@ -0,0 +1,70 @@
+  return (
+    <div className="WeeklyComponent">
+      <h2>Weekly Forecast</h2>
+      <ul className="WeeklyComponentList">
```
`className={styles.WeeklyComponentList}`

Shingo023 commented on src/components/WeeklyComponent.tsx:
```diff
@@ -0,0 +1,70 @@
+          days.map((day, index) => (
+            <li className="WeeklyComponentItem" key={index}>
+              <p className="WeeklyComponentDay">{getWeekday(day.datetime)}</p>
+              <p>icon</p>
```
`className={styles.WeeklyComponentItem}`
`className={styles.WeeklyComponentDay}`

Shingo023 commented on src/actions/weather.ts:
```diff
@@ -1,9 +1,18 @@
 import { WeatherData } from "@/types";
 
-export const getWeatherINfo = async (cityName: string) => {
+let currentCity: WeatherData;
```
How about `displayedCityWeather` or something as the name of the let variable?
Since we also have to deal with the current location and its weather somewhere else, `currentCity` might sound a bit confusing.

Shingo023 commented on src/actions/weather.ts:
```diff
@@ -1,9 +1,18 @@
   return data;
 };
+
+export const getCurrentCity = () => {
```
How about `getDisplayedCityWeather`?

Shingo023 reviewed (approved):
Your code was very clean and readable!
I just left some comments related to naming of variables and class names.

If you are OK with them, I'd like you to modify them and merge your PR. I'm approving it now.

I appreciate your great work and patience!

phfernandesmc committed:
Update server action weather.ts

Change the function and variable to fit better in the project

phfernandesmc committed:
Update WeeklyComponent.tsx

here I also changed some variables name fit well in the project and I imported the scss module for the styling.

Shingo023 reviewed (approved):
Thanks for the modifications!

Saeka03 commented on src/components/WeeklyComponent.tsx:
```diff
@@ -0,0 +1,70 @@
+import { useCurrentCity } from "@/contexts/currentCity";
+import { WeatherDay } from "@/types";
+import styles from "./WeeklyComponent.module.scss";
```
I think the path here would be like this?
```suggestion
import styles from "../style/components/WeeklyComponent.module.scss";
```

Saeka03 commented on src/components/WeeklyComponent.tsx:
```diff
@@ -0,0 +1,70 @@
+      <h2>Weekly Forecast</h2>
+      <ul className={styles.WeeklyComponentList}>
+        {weeklyWeather ? (
+          dailyWeather.map((day, index) => (
```
Is this "dailyWeather" declared somewhere? Or you mean, like this?
```suggestion
          weeklyWeather.map((day, index) => (
```

Saeka03 commented on src/components/WeeklyComponent.tsx:
```diff
@@ -0,0 +1,70 @@
+
+    const dayIndex = date.getDay();
+
+    console.log(dateString);
```
Is this console.log for debugging?
If it's unnecessary, can you delete it?

Saeka03 commented on src/components/WeeklyComponent.tsx:
```diff
@@ -0,0 +1,70 @@
+        {weeklyWeather ? (
+          dailyWeather.map((day, index) => (
+            <li className={styles.WeeklyComponentItem} key={index}>
+              <p className={styles.WeeklyComponentDay}>{getWeekday(day.datetime)}</p>
```
According to Aoi's design, if it's today, the day should be "Today"?
But we can modify it after merging it!
![image](https://github.com/Shingo023/weather-app/assets/143541044/a7c60251-6d44-4224-90b6-9bebae6425be)
![image](https://github.com/Shingo023/weather-app/assets/143541044/22da7dc3-fd6a-4f00-8424-e35b3390b66b)

Saeka03 commented on src/style/components/WeeklyComponent.css:
```diff
@@ -0,0 +1,28 @@
+
+.WeeklyComponentDay {
+  width: 90px;
+}/*# sourceMappingURL=WeeklyComponent.css.map */
```
Is this css file used somewhere?

Saeka03 commented on src/style/components/WeeklyComponent.css.map:
```diff
@@ -0,0 +1 @@
+{"version":3,"sources":["WeeklyComponent.scss","WeeklyComponent.css"],"names":[],"mappings":"AAGA;EACE,yBAAA;EACA,gBAAA;EACA,aAAA;EACA,mBAAA;ACFF;ADIE;EACE,gBAAA;EACA,mBAAA;ACFJ;;ADMA;EACE,gBAAA;EACA,aAAA;EACA,sBAAA;EACA,SAAA;ACHF;;ADMA;EACE,aAAA;EACA,8BAAA;EAEA,oBAAA;EACA,2CAAA;ACJF;;ADOA;EACE,WAAA;ACJF","file":"WeeklyComponent.css"}
```
Is this css file used somewhere?

Saeka03 commented on src/style/components/WeeklyComponent.scss:
```diff
@@ -0,0 +1,33 @@
+
+.WeeklyComponentDay {
+  width: 90px;
+}
```
I think the file name should be "WeeklyComponent.module.scss" to use scss in the WeeklyComponent.tsx file.

Saeka03 reviewed (approved):
Thank you for implementing! And I'm so sorry for being late to review your code.
I'll approve it and I'd appreciate it if you check my comments.

phfernandesmc committed:
updated the variables names and scss file name

Saeka03 reviewed (approved):
LGTM! Thank you for modifying it 😄
